modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheSharedTtlCleanupManager.java

```diff
@@ -155,6 +157,16 @@ private void stopCleanupWorker() {
 
                 try {
                     cctx.discovery().localJoin();
+
+                    try {
+                        cctx.exchange().affinityReadyFuture(AffinityTopologyVersion.ZERO).get();
+                    }
+                    catch (IgniteCheckedException ex) {
+                        log.error("Exception happened during waiting for topology initialization.", ex);
+
+                        throw new IgniteInterruptedCheckedException("Failed to wait for initialization topology [err="
```

Comment:
I don't think that we need to throw `IgniteInterruptedCheckedException` here.
It seems to me, it should be `IgniteException` as follows:
`
    catch (IgniteCheckedException ex) {
        throw new IgniteException("Failed to wait for initialization topology [err=" + ex.getMessage() + ']', ex);
    }
`